Agents/AirQualityAgent/airquality/dataretrieval/stations.py

```diff
@@ -0,0 +1,267 @@
+    #
+    # Initialise time series client   
+    ts_client = TSClient.tsclient_with_default_settings()
+    # Create output files for each set of retrieved time series data
+
+    print('Creating output files (geojson, metadata, timeseries) ...')
+    #logger.info('Creating output files (geojson, metadata, timeseries) ...')
+    t1 = time.time()
+
+    # 1) Create GeoJSON file for ReportingStations
+    stations = station_details[~station_details['dataIRI'].isna()]
+    geojson.append(create_geojson_output(stations, color, opacity))
+
+    # 2) Create JSON file for ReportingStations metadata
+    metadata.append(create_metadata_output(stations))
+
+    # 3) Create Time series output    
+    # Get List of corresponding dtvf ids for list of time series
+    # (to assign time series output to correct station in DTVF)
+    dataIRIs = [t.getDataIRIs()[0] for t in ts_data]
+    id_list = [int(stations.loc[stations['dataIRI'] == i, 'dtvf_id'].values) for i in dataIRIs]
+    tsjson = ts_client.convertToJSON(ts_data, id_list, ts_units, ts_names)
+    # Make JSON file readable in Python
+    timeseries = json.loads(tsjson.toString())
+    t2 = time.time()
+    diff = t2-t1
+    print(f'Finished after: {diff//60:5>n} min, {diff%60:4.2f} s \n')
+    #logger.info('Output files successfully created.')
+
+    # Create output files for stations without any time series data
+    stations = station_details[station_details['dataIRI'].isna()]
+    geojson.append(create_geojson_output(stations, color, opacity))
+    metadata.append(create_metadata_output(stations))
+
+    #
+    ###---  Write output files  ---###
+    #
+    print('Writing output files ...')
+    #logger.info('Writing output files ...')
+    for i in range(len(fp_geojson)):
+        with open(fp_geojson[i], 'w') as f:
+            json.dump(geojson[i], indent=4, fp=f)
+        with open(fp_metadata[i], 'w') as f:
+            json.dump(metadata[i], indent=4, fp=f)
+    # No time series data for stations w/o time series
+    with open(fp_timeseries, 'w') as f:
+        json.dump(timeseries, indent=4, fp=f)
+    print('Finished!\n')
+
+
+if __name__ == '__main__':
+
+    # Create station and time series output files
+    create_json_output_files('C:\TheWorldAvatar-git\Agents\AirQualityAgent\output')
+
+    # create_json_output_files('C:\TheWorldAvatar-git\Agents\AirQualityAgent\output',
+    #                          observation_types=['PM10Concentration'])
+
+    # Data retrieval with geospatial search require a geospatially-enabled
+    # Blazegraph namespace to work successfully
+    # create_json_output_files('C:\TheWorldAvatar-git\Agents\AirQualityAgent\output',
+    #                          circle_center='52.75#0.4', circle_radius='100')
+
+    # create_json_output_files('C:\TheWorldAvatar-git\Agents\AirQualityAgent\output',
+    #                          circle_center='52.75#0.4', circle_radius='100',
+    #                          observation_types=['PM10Concentration'])
```

Comment:
I assume this block is just for debugging/manual testing, is that correct?

Reply:
Yes, this code is only reached when executed as main script (i.e. called manually)